decorator.rb

```diff
@@ -0,0 +1,12 @@
+require './nameable'
+
+class Decorator < Nameable
+  def initialize(nameable)
+    @decoration = nameable
+    super()
```

Comment:
- Optional.
You can remove the super() as it is not necessary as the `Nameable class` does not have an explicit constructor.